### DISCUSSION TITLE
How to update flux-system repository from the CLI???

### DISCUSSION THREAD

**Discussion by CompPhy** [fluxcd/flux2#4915] [Q&A]
Maybe I'm just not looking in the right place, can someone help me understand here???  I guess my question is this... Is there a way to update the flux-system repository information from the CLI???  That being the URL and/or any related credentials.

Private repositories are fairly common and come with a number of constraints on how and where the are accessed.  Maybe the repository needs to move to a new server, or a token expired and needs to be manually renewed.  It seems like Flux could run into problems here, especially in emergency failure scenarios.  For example, maybe a server dies and backups need to be recovered to another location.

The reason I think this is a problem is because I have yet to see any way to update the main flux-system URL from the CLI.  In a failure of the git repository, at best Flux just can't do anything.  At worst it's conflicting with your ability to fix other issues, unless you explicitly suspend it's operation.

Even something as simple as pointing it to a different URL should be an option.  I do see that several different PR's have been opened for this over the years, but it would appear that none of them ever got implemented.  One would at least hope that the "troubleshooting" page in your documentation gives some recovery options if/when the git repository becomes inaccessible and needs some action to get it working again.

I'm also a little concerned that "uninstall" and "bootstrap fresh" would cause other problems.  Maybe the cluster is functioning fine and the only issue is the underlying git repository.   Isn't there a chance that bootstrapping again will cause it to redeploy all the services??  Which could end up inadvertently breaking something??  Maybe the intention is that flux sees all the existing deployments and just takes them over without re-deploying???  If that's the case, maybe there should be something in the documentation to state this as a possible recovery option?

**Comment by CompPhy** [comment]
I think I managed to answer my own question here.... It's not straight forward but does seem to work.  You can manually `kubectl apply -f flux-git.yaml -n flux-system` using something like the below example.  Maybe we can get an example like this into the documentation for troubleshooting purposes?

This does get overwritten as soon as flux reads the config from the repository itself, but in a pinch it's a hackish work around to get something working.

```yaml
apiVersion: source.toolkit.fluxcd.io/v1
kind: GitRepository
metadata:
  name: flux-system
  namespace: flux-system
spec:
  interval: 1m0s
  ref:
    branch: master
  secretRef:
    name: flux-system
  url: <REDACTED>

```

**Comment by stefanprodan** [comment]
> Even something as simple as pointing it to a different URL should be an option.

Have you tried rerunning bootstrap for the repo and it didn't worked?
